src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -217,7 +217,7 @@ uct_ib_mlx5_set_dgram_seg(struct mlx5_wqe_datagram_seg *seg,
 
     if (grh_av) {
         ucs_assert(av->dqp_dct & UCT_IB_MLX5_EXTENDED_UD_AV);
-#if HAVE_STRUCT_MLX5_GRH_AV_RMAC
+#if HAVE_STRUCT_MLX5_GRH_AV_RMAC || !HAVE_STRUCT_MLX5_WQE_AV_BASE
```

Comment:
why it's a problem if HAVE_STRUCT_MLX5_WQE_AV_BASE is defined?

Reply:
struct mlx5_wqe_av defined differently in mlx5dv.h and mlx5_hw.h
we have compat layer to work with it transparently - see https://github.com/openucx/ucx/blob/master/src/uct/ib/mlx5/ib_mlx5.h#L72
HAVE_STRUCT_MLX5_WQE_AV_BASE means mlx5_hw.h case.

Reply:
so in mlx5_hw.h we don't need to copy rmac? who sets it then?

Reply:
maybe in ib_mlx5.h, define HAVE_STRUCT_MLX5_GRH_AV_RMAC to 1 in mlx5_hw.h case?

Reply:
mlx5_hw.h case (HAVE_STRUCT_MLX5_WQE_AV_BASE defined) - in this case we need to check HAVE_STRUCT_MLX5_GRH_AV_RMAC, it was added to ofed at some point.
mlx5dv.h case (HAVE_STRUCT_MLX5_GRH_AV_RMAC not defined) - we define struct, not need to check

Reply:
i think force-defining HAVE_STRUCT_MLX5_GRH_AV_RMAC after we define the struct will be more self-explanatory than !HAVE_STRUCT_MLX5_WQE_AV_BASE